WordPress/gutenberg
# Add author and status filter to the page list

Related https://github.com/WordPress/gutenberg/issues/55083

## What?

Allow filtering by author and status:

https://github.com/WordPress/gutenberg/assets/583546/50c3198d-bec1-468a-a318-c299d934ec19

## Testing Instructions

- Add a couple of authors to the test site. Have them use a nice name by setting the "Display name".

<img width="561" alt="Captura de ecrã 2023-10-11, às 17 29 52" src="https://github.com/WordPress/gutenberg/assets/583546/2a01808d-deac-4e05-89a7-9827df310eaa">

- Create some pages for each. Make sure the pages have different status (publish, draft, scheduled).
- Enable the "new admin views" experiment.
- Go to "Site editor > Pages > Manage all pages".

Test the filter works as expected. Only the pages with `publish` and `draft` statuses are shown, as before.

## Next

Follow-ups (can be done in parallel):

- Allow user to change the visible filters.
- Allow multi-selection for `enumeration` filters (`author` and `status`).
- Look into adding `NOT IN` filters (the current ones are `IN` filters) at the top-level. For example, use `author_exclude`.
- Allow the `search` filter to be limited to one of `post_title`, `post_excerpt`, `post_content`. At the moment, it searches in all them.
- Add column filters (see [exploration](https://github.com/WordPress/gutenberg/pull/55270#issuecomment-1761415105)).
- Assess how the `search` could work differently. https://github.com/WordPress/gutenberg/pull/55475
- Use all post statuses, not just `publish` and `draft`. https://github.com/WordPress/gutenberg/pull/55476
